### DISCUSSION TITLE
Column values in tables are truncated

### DISCUSSION THREAD

**Discussion by kmdelucaNG** [Budibase/budibase#3588] [Help]
Budibase v0.9.168.

Is there a way to display more characters for table column values?  I have plenty of room in my table, but the values are truncated with "...".
Thanks.
![TruncatedValues](https://user-images.githubusercontent.com/86023779/143932782-7dcc1ab3-69e4-4380-b948-970e5327765a.jpg)

**Comment by mjashanks** [comment]
Unfortunately not. We have a bit of work to do with the table component: https://github.com/Budibase/budibase/discussions/1913

Instead of using a `Data Provider + Table`, you could try
- A `A Data Provider > Repeater > Container (Direction: Row)`
- Add components for every field manually, and set their width manually so that everything is aligned.

**Reply by nivbe06** [reply]
Hi @mjashanks - is there any update on this?
Thanks!

**Comment by danrenfroe2016** [comment]
@mjashanks  Hi, is this a priority to fix? Thanks.

**Reply by danrenfroe2016** [reply]
@mjashanks it appears to be fixed, and if not fixed, its much better!!! Thank you!!!

**Reply by MrChadMWood** [reply]
I see no way to adjust the number of characters before truncation occurs.

**Reply by melohagan** [reply]
Hey @MrChadMWood You can set the width of table columns as shown here: https://docs.budibase.com/docs/table#width

**Comment by atticuss-sra** [comment]
I still seem to be having an issue with this. Data is getting truncated far earlier than column width should require when using a Table Block component.

![image](https://github.com/Budibase/budibase/assets/135625870/00379ddc-6459-4189-9c47-e48355159809)

Dug into this a bit and it appears to be due to a `max-width` property being set. If you disable it, content displays as you would expect. 

![image](https://github.com/Budibase/budibase/assets/135625870/78e9fac5-2ca6-423e-b635-f521329bc1a6)

![image](https://github.com/Budibase/budibase/assets/135625870/91ccfb9d-b155-45d9-b3de-86b8e2106f11)

The `--max-cell-width` variable appears fixed to 200px and I can't find any way to change it. Tried overwriting the `max-width` prop using custom CSS on both the Table Block itself and page-wide via the Embed component. Even marking as important and tying to the `spectrum-Table-cell` class didn't seem to resolve it. That's not surprising though given that `max-width` is being set inline on a child div of that component. And as it doesn't have an ID or deterministic class, I'm not sure how to target it with CSS.

![image](https://github.com/Budibase/budibase/assets/135625870/a9bbe6ed-c037-41af-827d-a13cc03906d2)

Diagnostic details:

```
{
  "browser": {
    "language": "en-US",
    "userAgent": "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/119.0",
    "platform": "Linux x86_64",
    "vendor": ""
  },
  "server": {
    "budibaseVersion": "2.13.42",
    "hosting": "docker-compose",
    "nodeVersion": "v18.19.0",
    "platform": "linux",
    "cpuArch": "x64",
    "cpuCores": 2,
    "cpuInfo": "Intel(R) Xeon(R) Platinum 8259CL CPU @ 2.50GHz",
    "totalMemory": "3.7810134887695312GB",
    "uptime": "7 day(s), 22 hour(s), 51 minute(s)"
  }
}
```

**Reply by atticuss-sra** [reply]
It's worth noting that the container repeater trick from @mjashanks works but would require a lot of elbow grease to implement all the nice out-of-the-box features of a Table Block. Would much rather use that as it implements just about everything I need in most use-cases and makes it far easier for other people who haven't dug into Budibase internals to easily spin up apps (i.e., it the follows the low-code ethos).

**Reply by atticuss-sra** [reply]
To close the loop: I forgot wildcard selectors are a thing. Can resolve this via an Embed with the following:

```
<style>
    .spectrum-Table-cell [class*="svelte-"] {
        max-width: 2000px !important;
    }
</style>
```

No idea how this may affect other things. YMMV.
